packages/angular/src/lib/legacy/router/page-router-outlet.ts

```diff
@@ -404,7 +406,8 @@ export class PageRouterOutlet implements OnDestroy, RouterOutletContract {
       }
     });
 
-    const navOptions = this.locationStrategy._beginPageNavigation(this.frame);
+    this.locationStrategy._beginPageNavigation(this.frame);
+    const navOptions = { ...defaultNavOptions, ...(this.router.getCurrentNavigation().extras || {}) } as ExtendedNavigationExtras;
```

Comment:
I like the idea of removing the local cache, but maybe this PR could actually remove the local cache, too. Rather than just avoiding its usage.

https://github.com/NativeScript/angular/blob/44f1bcf917b5f92ed315d297274a76abdaac50aa/packages/angular/src/lib/legacy/router/ns-location-strategy.ts#L360
Unless there's a need to continue caching this (I don't see why there would be, as this should be able to also pull the .extras).